src/__init__.py

```diff
@@ -87,6 +87,25 @@ def root(netid):
 	return render_template("home.html", netid=netid)
 
 
+@app.context_processor
+def inject_header_values():
+	netid = auth.get_netid()
+	is_staff = netid and common.is_staff(netid)
```

Comment:
This `common.is_staff` checks if a person is a staff for **any** course. Suppose a user is course staff for course A but a student for course B, then he will see the "Student" button when he is on the course B's student page. When he clicks on it, he will be redirected to a forbidden page and see a 403. 

I can think of two ways to avoid this. One is to not show the "student" button the above happens. Another solution is to make this button not clickable.

Reply:
If I hide the "student" button, then they'd have to go through the Broadway logo to switch to the staff home later. Maybe a good compromise would be to switch them to staff_home instead of the staff course page for that course?

Reply:
I meant to say only hide the student button when 
1. The current user is a student for current course but not a staff
2. The current user is a staff for some other course
3. The current user is on the course/assignment page of the current course

In this case, the user should not be able to switch staff page of the current course at all. Try this out by adding a second course to your database that contains "staff" as a student.

Reply:
Alright! I'll keep in mind that student thing too and make sure to switch them to student_home instead as well!